### PR TITLE
Fix/dlq avoid to delete not segment files

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

[rn:skip]

## What does this PR do?

Restrict the DLQ segment deletion to only files that are considered segments.

## Why is it important/What is the impact to the user?

Avoid an exception in the down stream pipeline when the writer is actively pushing data.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files (and/or docker env variables)~~
- [x] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [x] run it in a real environment

## How to test this PR locally

This bug emerged while testing https://github.com/logstash-plugins/logstash-input-dead_letter_queue/pull/43 so it's difficult to reproduce on command line, using just external interaction.
To prove it, checkout this and revert the line  https://github.com/elastic/logstash/pull/14274/files#diff-22e56a5df12549ecbf9622646f8aa68040a9d95cda37f5376f8d2f1dbe790358L334 and run the unit test.

## Related issues

- Relates #14188

### CODE REVIEW

**Comment by andsel** [comment]
Jenkins test this please

**Review by mashhurs — APPROVED** [review]
LGTM!

File: logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java [hunk 1]
```diff
@@ -331,7 +331,7 @@ private void removeSegmentsBefore(Path validSegment) throws IOException {
         final Comparator<Path> fileTimeAndName = ((Comparator<Path>) this::compareByFileTimestamp)
                 .thenComparingInt(DeadLetterQueueUtils::extractSegmentId);
 
-        try (final Stream<Path> segmentFiles = Files.list(queuePath)) {
+        try (final Stream<Path> segmentFiles = DeadLetterQueueWriter.getSegmentPaths(queuePath)) {
```
Comment (robbavey) [inline logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java:334]: We can remove the `DeadLetterQueueWriter.` from here - we have a static import for this method, and we're using it elsewhere

Comment (andsel) [reply logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java:334]: Have aligned this in commit https://github.com/elastic/logstash/pull/14256/commits/fc2a2f9a01399a6d2c0bd09cb0d4313d968affa7

**Pull request merged**